src/Workspaces/Core/Portable/Shared/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -55,6 +55,20 @@ public static bool IsSystemVoid([NotNullWhen(returnValue: true)] this ITypeSymbo
         public static bool IsNullable([NotNullWhen(returnValue: true)] this ITypeSymbol? symbol)
             => symbol?.OriginalDefinition.SpecialType == SpecialType.System_Nullable_T;
 
+        public static bool IsNullable(
```

Comment:
I admit a GetUnderlyingValueTypeOfNullable might have been a bit clearer to me -- this smells very much of the "let's use a random out var to shuttle data around"... :-)

Reply:
It didn't seem random to me. This follows a pattern I'm used to with helper extension methods:

```cs
if (someType.IsIEnumerable(out var elementType))
{
```

```cs
if (someType.IsIReadOnlyDictionary(out var keyType, out var valueType))
{
```

And also the way I currently write DIY discriminated unions:

```cs
public readonly struct Result<T>
{
    [AllowNull, MaybeNull]
    private readonly T value;
    private readonly string? errorMessage;

    private Result<T>([AllowNull, MaybeNull] T value, string? errorMessage)
    {
        this.value = value;
        this.errorMessage = errorMessage;
    }

    public static Result<T> Success(T value)
    {
        return new Result<T>(value, errorMessage: null);
    }

    public static Result<T> Error(string message)
    {
        if (message is null) throw new ArgumentNullException(nameof(message));
        return new Result<T>(value: default, errorMessage: value);
    }

    public bool IsSuccess([MaybeNull] out T value)
    {
        value = this.value;
        return errorMessage is null;
    }

    public bool IsError([NotNullWhen(true)] out string? message)
    {
        message = errorMessage;
        return message is object;
    }

    // ...
}
```

Reply:
...but I won't take it personally if you clean it up :D

Reply:
Well, if I cared enough I would have made merging conditional on it. ;-)